pkg/vdri/trustbloc/didconfiguration/helpers.go

```diff
@@ -68,3 +72,141 @@ func createDomainLinkageAssertion(
 		JWT: jwsCompact,
 	}, nil
 }
+
+// VerifyDIDConfiguration verifies a DID configuration, using the given VDRI to resolve the DID.
+//   returns a list of the DIDs that were successfully authenticated to this domain
+func VerifyDIDConfiguration(domain string, configuration *models.DIDConfiguration, doc *did.Doc) ([]string, error) {
+	didSet := map[string]struct{}{}
+
+	var errs []string
+
+	for i, dla := range configuration.Entries {
+		err := ValidateDomainLinkageAssertion(domain, dla, doc)
+		if err != nil {
+			log.Debugf("domain linkage assertion %v for %s invalid", i, domain)
+
+			errs = append(errs, err.Error())
+		} else {
+			didSet[dla.DID] = struct{}{}
+		}
+	}
+
+	dids := make([]string, 0, len(didSet))
+	for id := range didSet {
+		dids = append(dids, id)
+	}
+
+	if len(dids) == 0 {
+		errMsg := ""
+		for _, ems := range errs {
+			errMsg += "`" + ems + "`, "
+		}
+
+		return nil, fmt.Errorf("all domain linkage assertions invalid for domain %s: %s", domain, errMsg)
+	}
+
+	return dids, nil
+}
+
+// ValidateDomainLinkageAssertion validates a domain linkage assertion, using the given VDRI to resolve the DID
+func ValidateDomainLinkageAssertion(domain string, assertion models.DomainLinkageAssertion, doc *did.Doc) error {
+	jws, err := jose.ParseSigned(assertion.JWT)
+	if err != nil {
+		return fmt.Errorf("cannot parse assertion JWT: %w", err)
+	}
+
+	var claims models.DomainLinkageAssertionClaims
+
+	rawClaims := jws.UnsafePayloadWithoutVerification()
+	err = json.Unmarshal(rawClaims, &claims)
+
+	if err != nil {
+		return fmt.Errorf("cannot parse assertion JWT claims: %w", err)
+	}
+
+	if claims.ISS != assertion.DID {
+		return fmt.Errorf("assertion DID does not match signature")
+	}
+
+	if claims.Domain != domain {
+		return fmt.Errorf("assertion domain does not match host domain")
+	}
+
+	if claims.Exp != 0 && claims.Exp <= time.Now().Unix() {
+		return fmt.Errorf("assertion has expired")
+	}
+
+	_, err = VerifyDIDSignature(jws, doc)
+
+	return err
+}
+
+// VerifyDIDSignature verify a signature using a DID doc
+func VerifyDIDSignature(jws *jose.JSONWebSignature, doc *did.Doc) ([]byte, error) {
+	if jws == nil {
+		return nil, fmt.Errorf("jws is nil")
+	}
+
+	jwkList := getJWKs(doc)
+
+	errs := ""
+
+	var val []byte
+
+	var err error
+
+	verified := false
+
+	for _, key := range jwkList {
+		if key == nil || key.Key == nil {
```

Comment:
How can key be nil in the list above? You are adding it in the helper function. Also, if key.Key is nil maybe you shouldn't add it the list in the first place.